rcl/test/rcl/test_count_matched.cpp

```diff
@@ -141,47 +83,109 @@ class TestCountFixture : public ::testing::Test
     delete this->context_ptr;
     EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
   }
+
+protected:
+  const rcl_guard_condition_t * graph_guard_condition;
+  rcl_context_t * context_ptr;
+  rcl_wait_set_t * wait_set_ptr;
+
+  void check_state(
+    rcl_publisher_t * publisher,
+    rcl_subscription_t * subscriber,
+    size_t expected_subscriber_count,
+    size_t expected_publisher_count,
+    size_t number_of_tries)
+  {
+    size_t subscriber_count = -1;
+    size_t publisher_count = -1;
+
+    rcl_ret_t ret;
+
+    for (size_t i = 0; i < number_of_tries; ++i) {
+      if (publisher) {
+        ret = rcl_publisher_get_subscription_count(publisher, &subscriber_count);
+        EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+        rcl_reset_error();
+      }
+
+      if (subscriber) {
+        ret = rcl_subscription_get_publisher_count(subscriber, &publisher_count);
+        EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+        rcl_reset_error();
+      }
+
+      if (
+        expected_publisher_count == publisher_count &&
+        expected_subscriber_count == subscriber_count)
+      {
+        RCUTILS_LOG_INFO_NAMED(ROS_PACKAGE_NAME, "  state correct!");
+        break;
+      }
+
+      if ((i + 1) == number_of_tries) {
+        // Don't wait for the graph to change on the last loop because we won't check again.
+        continue;
+      }
+
+      ret = rcl_wait_set_clear(wait_set_ptr);
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+      ret = rcl_wait_set_add_guard_condition(wait_set_ptr, graph_guard_condition, NULL);
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+      std::chrono::nanoseconds time_to_sleep = std::chrono::milliseconds(200);
+      RCUTILS_LOG_INFO_NAMED(
+        ROS_PACKAGE_NAME,
+        "  state wrong, waiting up to '%s' nanoseconds for graph changes... ",
```

Comment:
```suggestion
        " state wrong, waiting up to '%s' nanoseconds for graph changes... ",
```

Reply:
Actually, we don't need the space at all.  `RCUTILS_LOG_INFO_NAMED` will automatically add a space.  So I removed both spaces, and also removed it in another call above in 8d468ca.